src/NuGet.Indexing/IndexDirectoryProvider/CloudIndexDirectoryProvider.cs

```diff
@@ -75,7 +79,24 @@ public bool Reload(IndexingConfiguration config)
 
             var stopwatch = Stopwatch.StartNew();
 
-            var sourceDirectory = new AzureDirectory(storageAccount, _indexContainerName);
+            AzureDirectory sourceDirectory;
+
+            if (string.IsNullOrEmpty(config.AzureDirectoryCachePath))
+            {
+                sourceDirectory = new AzureDirectory(storageAccount, _indexContainerName);
+            }
+            else
+            {
+                _logger.LogInformation("Using custom azure directory cache path: {AzureDirectoryCachePath}", config.AzureDirectoryCachePath);
```

Comment:
I would either also log the absolute path with `Path.GetFullPath`, pass the full path to `AzureDirectory` ctor, or require `Path.IsPathRooted`.

As it stands today, a configured relative path has unclear behavior.

Reply:
FYI, `IsPathRooted` does not guarantee path is absolute. Try
    
     [System.IO.Path]::IsPathRooted('c:a.txt');

in powershell.

Reply:
Oh good point. In this case, I would just avoid `IsPathRooted`...

Reply:
I went ahead and did `Path.GetFullPath`